cmake/ProjUtilities.cmake

```diff
@@ -98,7 +98,7 @@ function(configure_proj_pc)
     list(APPEND EXTRA_LIBS -lole32 -lshell32)
   else()
     set(cxx_libs "${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES}")
-    list(REMOVE_ITEM cxx_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES})
+    list(REMOVE_ITEM cxx_libs "${CMAKE_C_IMPLICIT_LINK_LIBRARIES}")
```

Comment:
This is the wrong fix.
We want to remove all individual list items in `CMAKE_C_IMPLICIT_LINK_LIBRARIES`, not the single item (or specific sequence) which is the concatenated value.
To work around the bug in old CMake versions, use a dummy item, i.e.
~~~cmake
list(REMOVE_ITEM cxx_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES} CMake_3.19_quirk)
~~~

Cf. https://github.com/OSGeo/gdal/pull/5363

Reply:
POC:
~~~cmake
set(list1 A B C)
set(list2 A C)
list(REMOVE_ITEM list1 "${list2}")
message(STATUS "${list1}")
list(REMOVE_ITEM list1 ${list2})
message(STATUS "${list1}")
list(REMOVE_ITEM list1 ${list3} CMake_3.19_quirk)
message(STATUS "${list1}")
~~~

CMake 3.19 output:
~~~
-- A;B;C
-- B
-- B
~~~

Reply:
what do you think of:

```cmake
if(CMAKE_C_IMPLICIT_LINK_LIBRARIES)
    list(REMOVE_ITEM cxx_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES})
endif()
```

Reply:
This is what is done in GDAL. I dislike it because it is longer but still hides why it is needed at all.

Reply:
Thanks @dg0yt for the useful example, I've modified this to the 1-liner, using a nonempty item at the end.